### ISSUE TITLE
Subtitles stretched on resolutions greater than 16:9

### ISSUE THREAD

**Issue by gajewsk2** [Kaldaien/TZF#18]
Using a 5760x1080 setup. Main UI elements like text boxes appear on my center screen and without stretching, but battle damage counter and cutscene subs stretch past the center screen and very stretched out. Is that normal or is there a setting I can tweak?

**Comment by Kaldaien** [comment]
Set AspectCorrection to true. It's diabled in 1.1.0 by default.

**Comment by gajewsk2** [comment]
It was set to true by default for me, here's mine:
```
[TZFIX.Audio]
Channels=8
SampleRate=48000
CompatibilityMode=false
EnableFix=true

[TZFIX.FrameRate]
SpeedResetCode_Address=4637993
SpeedResetCode2_Address=5694529
SpeedResetCode3_Address=5691711
LimiterBranch_Address=10029171
AllowFakeSleep=false
AllowWindowedMode=true
YieldProcessor=true
FixStutter=false
FudgeFactor=1.666666
MinimizeLatency=false
DisableNamcoLimiter=false
AutoAdjust=false
Target=60

[TZFIX.Steam]
AllowBroadcasts=false

[TZFIX.Render]
AspectRatio_Address=13984932
FOVY_Address=13984936
AspectCorrectVideos=true
AspectCorrection=true
CompleteMipmaps=false
PostProcessRatio=1.000000
RescaleShadows=0
RescaleEnvShadows=0
DisableScissor=false
AspectRatio=5.333333
FOVY=0.785398

[TZFIX.System]
Version=1.1.0
```

**Comment by drdaxxy** [comment]
If you're playing on the latest version of the game and you're still running TZFix 1.1.0, most features will be broken. Try with 1.1.1.

That said, I have also noticed some elements sometimes being stretched or displayed at the wrong locations on extreme aspect ratios, even when the aspect ratio fix otherwise works. It's not perfect.

**Comment by gajewsk2** [comment]
I was on 1.3, so shouldn't be an issue since everything else works. Alright, I'll try updating and post the results.

**Comment by Kaldaien** [comment]
I'm still working on this, but it's proving to be a real challenge. I cannot for the life of me get the bloom filter separated from the UI. I can get everything in the UI scaled correctly but then some post-processing effects go awry.

You might check the 1.1.3 test0 release I pushed earlier, it will very likely create new artifacts on the sides of the screen, but should scale most things properly.

**Comment by gajewsk2** [comment]
Thanks, I'll check it out. I know I'm not alone appreciating all the effort going into this.

**Comment by gajewsk2** [comment]
![zestiria stretched](http://imageshack.com/a/img633/5127/V0er4R.png)

Seems the vertical ratio got really messed up with the new update...

**Comment by Kaldaien** [comment]
I believe this has been fixed in 1.1.4 Test0.

https://github.com/Kaldaien/TZF/releases/tag/tzf_114-test0

If I could get test feedback on mouse coordinates and the new system that clears the framebuffer on the tops/sides that exceed 16:9 aspect ratio, that'd be a big help and I will consider this issue closed.

**Comment by gajewsk2** [comment]
Ok, so newest release: Damage counters are just right now, great job. 
Subs are still stretched for me though. Do you need a screenshot?

**Comment by Kaldaien** [comment]
Yes, a screenshot would be helpful. I cannot seem to reproduce this problem.

**Comment by gajewsk2** [comment]
![zestiria subs](http://imageshack.com/a/img907/1859/KudhZ8.png)

**Comment by Kaldaien** [comment]
I see, thank you. It looks like I have not caught all of the vertex shaders responsible for drawing the UI.

I've definitely got a good 90+% of them working as intended.


Can I ask you, does your INI file still read this way:

    AspectRatio=1.333333


The resolution you described has a 5.33333 aspect ratio, and TZfix is setup to write the aspect ratio that it computes to the INI file. It looks like it is not computing your aspect ratio correctly.

**Comment by gajewsk2** [comment]
```
[TZFIX.Audio]
Channels=8
SampleRate=48000
CompatibilityMode=false
EnableFix=true

[TZFIX.FrameRate]
FixStutter=false
FudgeFactor=1.666000
AllowFakeSleep=false
YieldProcessor=true
AllowWindowedMode=true
MinimizeLatency=false
SpeedResetCode_Address=4636921
SpeedResetCode2_Address=5696321
SpeedResetCode3_Address=5693503
LimiterBranch_Address=10030931
DisableNamcoLimiter=true
AutoAdjust=false
Target=61

[TZFIX.Render]
AspectRatio_Address=13984916
FOVY_Address=13984920
AspectRatio=1.333333
FOVY=0.785398
AspectCorrectVideos=true
AspectCorrection=true
CompleteMipmaps=false
PostProcessRatio=1.000000
RescaleShadows=-2
DisableScissor=false
RescaleEnvShadows=0

[TZFIX.Steam]
AllowBroadcasts=false

[TZFIX.System]
Version=1.1.4
IntroVideo=RAW\MOVIE\AM_TOZ_OP_001.BK2
```

**Comment by Kaldaien** [comment]
I am able to reproduce this problem now, and I will do a detailed shader dump tomorrow and see if I can correct it. Just wanted to give you an update so you didn't think I forgot about you ;)

The release of 1.2.x has been rocky and that has occupied most of my free time.

**Comment by gajewsk2** [comment]
😆 awesome! thanks so much for the update.  cant wait to try it!

**Comment by Kaldaien** [comment]
I know I said tomorrow, but I had an easier time than I expected =P

Grab and rename tzfix-subs.dll to tzfix.dll, https://github.com/Kaldaien/TZF/releases/tag/tzf_122

Give that a whirl and let me know if it breaks anything ;)

**Comment by gajewsk2** [comment]
Definitely fixed the issues I ran into. Question though, when you open a treasure chest (or other ui stuff like get a message about how something works) the center monitor in my case shows everything perfect. The two side monitors go black though. So basically, right before you open a treasure everything looks good at 5780x1080, then the game goes smoothly to 1920x1080, and then comes out of the treasure popup and back to original resolution. Seems like that might be intentional on your end, but still a little strange. Is there anyway to keep the sides from going black in a case like this?

**Comment by Kaldaien** [comment]
Yes, that is intentional. That behavior is covered by the ini option:

`ClearBlackbars`

For any gamestate that correlates to a UI where the sides of the non-16:9 sides of the screen contain garbage, when that option is set to true it clears them. There are a few false positives and a few cases that are not caught, unfortunately.

It catches most of them and generally is not _too_ intrusive. Without that option, you'd see purple garbage on the sides of the screen when the main menu is opened. I may introduce an option to just clear the sides on the main menu and during loading, since that's where it is worst.

<hr>

I'll leave this open, you've been very helpful and I'll think of a better solution for the blackbar problem for you to test eventually ;)

**Comment by gajewsk2** [comment]
Ahh thanks for that info. Will do!

**Comment by Kaldaien** [comment]
Most of the stuff discussed has been addressed in 1.2.3.

The last remaining issue is the targeting indicators in battle. They do not scale correctly at the moment. I will try to fix this in January.
